src/execution/operator/schema/physical_create_index.cpp

```diff
@@ -111,11 +109,14 @@ SinkFinalizeType PhysicalCreateIndex::Finalize(Pipeline &pipeline, Event &event,
 	// here, we just set the resulting global index as the newly created index of the table
 
 	auto &state = (CreateIndexGlobalSinkState &)gstate_p;
-
 	if (!table.storage->IsRoot()) {
 		throw TransactionException("Transaction conflict: cannot add an index to a table that has been altered!");
 	}
 
+	if (state.global_index->track_memory) {
+		state.global_index->buffer_manager.IncreaseUsedMemory(state.global_index->memory_size);
```

Comment:
The `IncreaseUsedMemory` seems to only be called in the `PhysicalCreateIndex` - this is not the code path that is taken for primary key indexes, and memory will also not be tracked/increased for newly inserted data. Perhaps this should be moved to the `Index::Insert` and `Index::Remove` calls instead, or also added to other locations where the index memory usage increases?

Reply:
To increase the ART size in the buffer manager, I added code to `ART::Insert` (`art.cpp`) and the `physical_create_index.cpp` . Basically directly after every time we change the ART, so for each chunk of inserted keys or each new ART. This should also work for primary keys and newly inserted data, no? I added code to decrease the memory usage to the destructor `ART::~ART()` and `ART::Delete` (`art.cpp`). If I remember correctly, these are all the places where I propagated changes to the buffermanager.

But yes, maybe these places are not intuitive enough? Or there is a better solution? I intended to change the memory usage chunk-wise directly after a permanent change to the ART.

Reply:
Ah yes, you are correct - I missed the usage in `ART::Insert`